fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -980,115 +972,15 @@ private void createJob(long dbId, OlapTable olapTable, Map<Long, LinkedList<Colu
             }
 
             // 5. calc short key
-            short newShortKeyColumnCount =
-                    GlobalStateMgr.calcShortKeyColumnCount(alterSchema, indexIdToProperties.get(alterIndexId));
-            LOG.debug("alter index[{}] short key column count: {}", alterIndexId, newShortKeyColumnCount);
-            indexIdToShortKeyColumnCount.put(alterIndexId, newShortKeyColumnCount);
+            short newShortKeyCount = GlobalStateMgr.calcShortKeyColumnCount(alterSchema, indexIdToProperties.get(alterIndexId));
+            LOG.debug("alter index[{}] short key column count: {}", alterIndexId, newShortKeyCount);
 
-            // 6. store the changed columns for edit log
-            changedIndexIdToSchema.put(alterIndexId, alterSchema);
+            jobBuilder.withNewIndexShortKeyCount(alterIndexId, newShortKeyCount).withNewIndexSchema(alterIndexId, alterSchema);
 
             LOG.debug("schema change[{}-{}-{}] check pass.", dbId, tableId, alterIndexId);
         } // end for indices
 
-        if (changedIndexIdToSchema.isEmpty() && !hasIndexChange) {
```

Comment:
The codes between line 994 and 1092 have been moved to method `OlapTableAlterJobV2Builder.build()`.